### ISSUE TITLE
[Feature Request] Programmatically Highlight or Scroll to Search Results

### ISSUE THREAD

**Issue by Halfist** [microsoft/monaco-editor#3377]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

In our project we use Monaco Editor (via [ngx-monaco-editor](https://github.com/atularen/ngx-monaco-editor) port for Angular) for config editing. There's a need to implement search and present results functionality.

There are two options that satisfy user.

#### First: search and highlight
_Multiple highlighting would be great, but if not, highlighting the first match also works._

E.g. after `monaco.editor.searchAndHighlight('world');`

![image](https://user-images.githubusercontent.com/5636672/196726535-450d8d0e-7b97-496e-a804-e9e3a78fb68b.png)

editor should look like

![image](https://user-images.githubusercontent.com/5636672/196722432-31ea7a1a-47fd-44ec-8a59-894cdecbb38d.png)

#### Second: search and scroll to
_Here I think we could only scroll to the first match._

E.g. after `monaco.editor.searchAndScrollTo('comment');`

![image](https://user-images.githubusercontent.com/5636672/196728370-d9385538-a8ea-4488-b1c8-b3772e537499.png)

editor should look like

![image](https://user-images.githubusercontent.com/5636672/196728679-cafdc814-6fc1-4923-8f17-7077cd8cd59d.png)

So far I could only find examples of syntax highlighting, but not this. So, I considered this functionality missing.

Basically, it's programmatical Ctrl+F, but with possibility to highlight multiple search results in the same time (by modifying HTML from `text` to `<span class="mark">text</span>` I guess).

**Comment by Halfist** [comment]
Search and highlight functionality appeared to be supported already, and [here](https://stackoverflow.com/a/74141067/1611569)'s how to use it.

The issue is closed.
